### PR TITLE
refactor(feel-scala): replace deprecated Builder with FeelEngineBuilder API

### LINKED ISSUE #863: Migrate ScalaFeelEngine to Use FeelEngineBuilder and FeelEngineApi
**What needs to be done? Please describe.**

The `ScalaFeelEngine` currently uses the deprecated `org.camunda.feel.FeelEngine.Builder` class to create the engine instance. This should be replaced with the modern `FeelEngineBuilder` API. Additionally, the return type must be updated from `FeelEngine` to `FeelEngineApi`, and all usages of the engine throughout the codebase must be updated accordingly.

Tasks:
- Replace usage of the deprecated `Builder` with `FeelEngineBuilder`.
- Change the return type from `FeelEngine` to `FeelEngineApi`.
- Refactor all usages of the engine to use the `FeelEngineApi` interface instead of the old implementation.

**Additional context**

Old (deprecated) code:
```java
protected org.camunda.feel.FeelEngine buildFeelEngine(CustomFunctionTransformer transformer,
                                                      CompositeValueMapper valueMapper) {
  return new Builder()
    .functionProvider(transformer)
    .valueMapper(valueMapper)
    .enableExternalFunctions(false)
    .build();
}
```

New code:
```java
protected FeelEngineApi buildFeelEngine(CustomFunctionTransformer transformer,
                                        CompositeValueMapper valueMapper) {
  return FeelEngineBuilder
    .forJava()
    .withFunctionProvider(transformer)
    .withValueMapper(valueMapper)
    .withEnabledExternalFunctions(false)
    .build();
}
```

### PR DESCRIPTION
**Pull request by ungerts**
### 📄 Summary

Refactored the Scala FEEL engine integration by replacing the deprecated `org.camunda.feel.FeelEngine.Builder` with the new `FeelEngineBuilder` and `FeelEngineApi`. This change ensures compatibility with the latest FEEL APIs and improves maintainability and clarity of the engine integration.

### 🔗 Related Issues

Closes #863

### 📌 Motivation and Context

The `FeelEngine.Builder` class has been deprecated and replaced by `FeelEngineBuilder` with the `FeelEngineApi` interface. Migrating to the newer API improves long-term supportability, aligns with current FEEL best practices, and removes the dependency on legacy internals.

### 🛠️ Implementation Details

- Replaced all usages of the deprecated `FeelEngine.Builder` with `FeelEngineBuilder.forJava()`.
- Refactored `ScalaFeelEngine` to internally use `FeelEngineApi` instead of `FeelEngine`.
- Updated method implementations to use the new `evaluateExpression` and `evaluateUnaryTests` APIs.
- Added extensive inline documentation to clarify responsibilities and enhance readability.
- Introduced helper methods for evaluation result handling and variable context resolution.
- Updated factory class `ScalaFeelEngineFactory` accordingly.
- Added a comprehensive test suite (`ScalaFeelEngineTest`) covering various FEEL evaluation scenarios including edge cases and real-world inputs.

### 🧪 How Was This Tested?

- ✅ Added 40+ unit tests covering:
  - Numeric, string, logical expressions
  - Unary tests and ranges
  - Complex expressions, edge cases, and built-in functions
  - Null safety and error handling
- ✅ Verified all tests pass with `mvn clean test`
- ✅ Manual validation of edge-case expressions

### 🖼️ Screenshots or Logs (if applicable)

N/A

### 💥 Breaking Changes?

No

---

#### 📋 Administrative
- [x] I have read and followed the [contribution guidelines](https://github.com/operaton/operaton/blob/main/CONTRIBUTING.md)
- [x] I confirm that my contribution complies with the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0)
- [x] This PR references an issue (Closes #863)

#### 🧪 Testing
- [x] Unit tests were written or updated
- [x] JUnit 5 and `ProcessEngineExtension` were used for engine tests
- [x] `mvn clean test` passes all unit tests

#### 🧹 Project Hygiene
- [x] Commits follow [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Branch is up to date with `main`
- [x] Manual verification performed

---

### 🧭 Change Type

🔨 Refactoring (no functional or API changes)

### CODE REVIEW

File: engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/DefaultDmnEngineConfiguration.java [hunk 1]
```diff
@@ -1,10 +1,9 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
- * under one or more contributor license agreements. See the NOTICE file
- * distributed with this work for additional information regarding copyright
- * ownership. Camunda licenses this file to you under the Apache License,
- * Version 2.0; you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
```
Comment (javahippie) [inline engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/DefaultDmnEngineConfiguration.java:7]: From a legal perspective: We consulted a lawyer before forking, and his advice regarding license headers was, that we are only allowed to change them if we made "substantial functional changes" to existing classes. Even in this case, he recommended leaving the old license headers in place, so I'd feel better if we only add Copyright for "the Operaton contributors" in newly created files

Comment (ungerts) [reply engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/DefaultDmnEngineConfiguration.java:7]: I added the "Modifications Copyright" line to files with clear functional changes, as required by Apache License 2.0 §4(b). Original headers were kept fully intact.

For new files, it's just the Operaton copyright, as discussed.

Comment (ungerts) [reply engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/DefaultDmnEngineConfiguration.java:7]: Thanks for the feedback! Additional Remark: I followed the approach that aligns with the legal guidance you mentioned:

- I kept the original Camunda copyright untouched in all existing files.
- For the parts I’ve added or changed, I included "Copyright 2025 The Operaton Contributors" – as an addition, not a replacement.

It’s not really about whether the changes are “substantial” – that’s a vague term legally. What matters is whether a change is protected by copyright, and that can be true even for smaller or more targeted contributions. Leaving out a copyright notice could imply that my work falls under Camunda’s copyright, which isn’t the case and isn’t what I intend. I’m contributing it to the Operaton project, not transferring it to Camunda.

That said, if we want to define a clearer convention for this kind of situation within the project, I’m totally open to that. But from a legal and contributor perspective, I believe this way is safe and transparent.

**Comment by javahippie** [comment]
Hi, sorry for the delayed response. To me this looks fine, thanks a lot!

**Comment by kthoms** [comment]
@ungerts I have rebased the branch and resolved the conflicts. Mainly re-ordered import statements.
